### ISSUE TITLE
Add non-exclusive transactions

### ISSUE THREAD

**Issue by slavat** [greenrobot/greenDAO#159]
I have quite long inserts inside transactions in my project (several seconds). SQLITE is blocked on read and write by default during transactions. So read requests are waiting for the DB while insert transactions are in progress.

Android has support for write-ahead logging and non-exclusive transactions since API 11. If you call SQLiteDatabase.enableWriteAheadLogging() and begin transactions with beginTransactionNonExclusive() you can execute reads while transactions are in progress.

It will be great to have support for non-exclusive transactions in AbstractDao and anywhere else if necessary.

**Comment by equreshi** [comment]
Is there an ETA on this feature?

**Comment by nickylin** [comment]
is anyone try to use this feature for non-exclusive transactions? will this work?

**Comment by carlpoole** [comment]
Non-exclusive transactions would be a big help here too. Thank you
